Non blocking

seem-less opened the pull request:
Implemented another thread responsible for sending frames to the DAC. Frames are sent to the thread through a channel, the thread then sends the first frame in the channel queue to the DAC when it's status is ready.

Tested using a Helios DAC and Laserworld CS-2000RGB-FX laser.

seem-less committed:
compiled

seem-less committed:
running

seem-less committed:
cleaned up

seem-less committed:
edited cargo.toml

maxjoehnk commented on src/native.rs:
```diff
@@ -68,6 +76,109 @@ pub enum NativeHeliosDac {
+impl NonBlockingNativeHeliosDac{
+    pub fn new(dac:NativeHeliosDac)->Self{
+        let (sender, receiver) = mpsc::channel();
+        let receiver = Arc::new(Mutex::new(receiver));
```
You don't need the `Arc<Mutex<>>` here as the receiver is only used in the new thread.

maxjoehnk commented on src/native.rs:
```diff
@@ -68,6 +76,109 @@ pub enum NativeHeliosDac {
 }
 
+pub struct NonBlockingNativeHeliosDac{
+    pub dac:Arc<NativeHeliosDac>,
```
Imo the dac shouldn't be exposed here as it should only expose a non blocking api.

maxjoehnk reviewed (changes requested):
Thanks for this PR!

I have more remarks regarding the design of this PR but I don't have enough time right now.
I would like to avoid any `panic!` calls in this library though and I also think the `DONT_BLOCK` flag is not necessary in the native implementation as we're exposing a different struct which will never block.

I would also prefer if you could undo the default feature change and version bump as I would prefer to keep the commits clean from other changes.
In my eyes the change of the default feature should also bump the "major" version as it's a breaking change (though I am aware that pre 1.0.0 everything can be a breaking change).

seem-less committed:
reviewer changes implemented

seem-less commented:
Hey Max, thanks for the quick feedback!

I have removed the panic calls, reverted the default feature change and removed the version bump. I have kept the WriteFrameFlags::DONT_BLOCK flag in the implementation because it seems that this data is sent with the frame to the DAC, and so I assume that the DAC MCU would need this data to manage it's internal buffer, although it is not clear if this is the case when looking at the [original SDK](https://github.com/Grix/helios_dac/blob/f2da9499a94fb71eb68b11c5051a48fac2c846a7/sdk/HeliosDac.h#L20).

If you have anymore suggestions with regards to the design of this PR, I would be happy to discuss and implement them
